qutebrowser/completion/completer.py

```diff
@@ -90,8 +90,8 @@ def _get_new_completion(self, before_cursor, under_cursor):
             log.completion.debug('Starting command completion')
             return miscmodels.command
         try:
-            cmd = objects.commands[before_cursor[0]]
-        except KeyError:
+            cmd = parser.CommandParser().parse_all(before_cursor[0])[0].cmd
```

Comment:
This works, though I wonder if it's overkill and we should just use `config.cache['aliases'][before_cursor[0]]`?

Reply:
+1, both of looking stuff up in objects.commands and config.cache['aliases'] are an order of magnitude faster than parsing a command line. And that's just what the [command parser does](https://github.com/qutebrowser/qutebrowser/blob/8ddaef3/qutebrowser/commands/parser.py#L60) to resolve aliases.

@knezi you should be able to plop a `config_stub.val.aliases = {"openalias": "open"}`, or whatever, in `test_update_completion` and add some more parameterized test entries.